### PR TITLE
Basic build configuration for fuzzing with AFL++

### PR DESCRIPTION
**Pull request by lukasz-tuz**
Commit adds simple infrastructure for fuzzing the framework's functions with AFL++. New build options are introduced:

- `fuzzing`: enables the fuzzing target build (default false)
- `afl_inc`: absolute path to directory containing AFL++ headers
- `coverage`: enables building with coverage compiler options

For convinience, `build-fuzzing.sh` script wraps meson/ninja commands for building the fuzzing target:

```bash
./scripts/build-fuzzing.sh <builddir> -Dafl_inc=/absolute/path/to/afl/include
```

`afl_inc` argument is optional, meson will attempt to locate the header in typical locations if `-Dafl_inc` is not passed.

Resulting binary contains a small set of fuzzing tests grouped in the `fuzzing` test group. Fuzzing can be started using standard AFL++ command line options, e.g.:

```bash
afl-fuzz -i <inputs> -o <outputs> -- <builddir>/opendcdiag --beta -e fuzzer_memcmp_or_fail @@
```

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -4,12 +4,12 @@
 project(
     'OpenDCDiag',
     ['c', 'cpp'],
-    default_options : [
+    default_options: [
```
Comment (thiagomacieira) [inline meson.build:7]: Clean ups are correct but noise here. Can you submit a separate patch for them?

File: meson.build [hunk 2]
```diff
@@ -264,33 +282,53 @@ executable(
...
+    native: true,
 )
 
-summary({
+if get_option('fuzzing') == true
+    executable(
+        'fuzzing',
+        install: false,
+        dependencies: target_deps,
+        link_with: [
```
Comment (thiagomacieira) [inline meson.build:308]: Why a separate executable, instead of the regular one?

File: tests/fuzzer/fuzzer.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/**
```
Comment (thiagomacieira) [inline tests/fuzzer/fuzzer.c:1]: Should this test live inside framework, like the selftests and MCE check?

**Review by thiagomacieira — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "C_Cpp.default.compileCommands": "/mnt/home/ltuz/dev/fork-opendcdiag/builddir/compile_commands.json",
+    "C_Cpp.default.configurationProvider": "mesonbuild.mesonbuild"
+}
```
[the remark this answers is not in the document]
Comment (lukasz-tuz) [reply .vscode/settings.json]: Yes, definitely `.vscode` needs to be added to `.gitignore`.

File: meson.build (hunk 1 above)
Comment (lukasz-tuz) [reply meson.build]: Yes, you're right, I need to refrain from auto-linting these files.

File: meson.build (hunk 2 above)
Comment (lukasz-tuz) [reply meson.build]: Seemed like a good idea at the time. Using the regular one makes it simpler though, so I'll remove the separate build target.

File: tests/fuzzer/fuzzer.c (hunk 3 above)
Comment (lukasz-tuz) [reply tests/fuzzer/fuzzer.c]: You're right, it is better suited inside framework.

**Review by thiagomacieira — APPROVED** [review]
Thanks. This looks good.

But I wonder if we should add it to the selftests instead of the regular test roster instead.

**Comment by lukasz-tuz** [comment]
> But I wonder if we should add it to the selftests instead of the regular test roster instead.

I'll look into it and submit a separate PR.
